QuickCheck.cabal

```diff
@@ -254,7 +254,7 @@ Test-Suite test-quickcheck-misc
     Default-language: Haskell2010
     hs-source-dirs: tests
     main-is: Misc.hs
-    build-depends: base, QuickCheck
+    build-depends: base, QuickCheck, containers, nothunks
```

Comment:
I don't mind having the test at all, that's a good thing. But perhaps we could have this test in its own tiny test suite? That way it will be easier to turn off without turning everything else off (if need be).